daemon/api_snaps.go

```diff
@@ -654,6 +660,41 @@ func snapUpdateMany(inst *snapInstruction, st *state.State) (*snapInstructionRes
 	}, nil
 }
 
+func snapEnforceValidationSets(inst *snapInstruction, st *state.State) (*snapInstructionResult, error) {
+	if len(inst.ValidationSets) > 0 && len(inst.Snaps) != 0 {
+		return nil, fmt.Errorf("snap names cannot be specified with validation sets to enforce")
+	}
+
+	snaps, ignoreValidation, err := snapstate.InstalledSnaps(st)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := assertstateRefreshSnapAssertions(st, inst.userID, nil); err != nil {
+		return nil, err
+	}
+
+	var validErr *snapasserts.ValidationSetsValidationError
+	err = assertstateTryEnforceValidationSets(st, inst.ValidationSets, inst.userID, snaps, ignoreValidation)
```

Comment:
nitpick: looking only at this funciton, I thought that `ignoreValidation` was a boolean flag. I think we can make it a bit more readable if we rename the variable; on top of my mind I'd suggest excludedSnaps, prevalidatedSnaps, uncheckedSnaps, ignoredSnaps.